### ISSUE TITLE
[Idea] Store Component Sets in the ECS

### ISSUE THREAD

**Issue by zicklag** [jcornaz/heron#127]
If I understand correctly from what I've seen in `bevy_rapier2d`, it looks like it actually stores all of the simulation data such as rigid body positions, etc. in the bevy ECS, by implementing the [`ComponentSet`](https://docs.rs/rapier2d/0.9.2/rapier2d/data/trait.ComponentSet.html) trait. With the way we have `heron` right now we are storing the physics world data in Bevy resources that use arena based storage.

I'm curious if that has any advantages performance or otherwise and if that would be a useful strategy in `heron` or not. I'm mostly just musing over the possibility, but do you have any ideas why we would want to or not want to do that?

**Comment by jcornaz** [comment]
Yes, that's something I had in mind as soon as I heard it would be possible with the new version of rapier.

In theory it should be as beneficial for heron as it is for `bevy_rapier`.
